components/script/dom/window.rs

```diff
@@ -589,8 +595,152 @@ impl<'a> WindowMethods for &'a Window {
+    fn ScrollTo(self, options: &ScrollToOptions) {
+        self.Scroll(options);
+    }
+
+    // https://drafts.csswg.org/cssom-view/#dom-window-scrollto
+    fn ScrollTo_(self, x: f64, y: f64) {
+        self.scroll(x, y, ScrollBehavior::Auto);
+    }
+
+    // https://drafts.csswg.org/cssom-view/#dom-window-scrollby
+    fn ScrollBy(self, options: &ScrollToOptions)  {
+        // Step 1
+        let x = options.left.unwrap_or(0.0f64);
+        let y = options.top.unwrap_or(0.0f64);
+        self.ScrollBy_(x, y, );
+        self.scroll(x, y, options.parent.behavior);
+    }
+
+    // https://drafts.csswg.org/cssom-view/#dom-window-scrollby
+    fn ScrollBy_(self, x: f64, y: f64)  {
+        // Step 3
+        let left = x + self.ScrollX() as f64;
+        // Step 4
+        let top =  y + self.ScrollY() as f64;
+
+        // Step 5
+        self.scroll(left, top, ScrollBehavior::Auto);
+    }
+
+    // https://drafts.csswg.org/cssom-view/#dom-window-resizeto
+    fn ResizeTo(self, x: i32, y: i32) {
+        // Step 1
+        //TODO determine if this operation is allowed
+        let size = Size2D::new(x.to_u32().unwrap_or(1), y.to_u32().unwrap_or(1));
+        self.compositor.send(ScriptToCompositorMsg::ResizeTo(size)).unwrap()
+    }
+
+    // https://drafts.csswg.org/cssom-view/#dom-window-resizeby
+    fn ResizeBy(self, x: i32, y: i32) {
+        let (size, _) = self.client_window();
+        // Step 1
+        self.ResizeTo(x + size.width.to_i32().unwrap_or(1), y + size.height.to_i32().unwrap_or(1))
+    }
+
+    // https://drafts.csswg.org/cssom-view/#dom-window-moveto
+    fn MoveTo(self, x: i32, y: i32) {
+        // Step 1
+        //TODO determine if this operation is allowed
+        let point = Point2D::new(x, y);
+        self.compositor.send(ScriptToCompositorMsg::MoveTo(point)).unwrap()
+    }
+
+    // https://drafts.csswg.org/cssom-view/#dom-window-moveby
+    fn MoveBy(self, x: i32, y: i32) {
+        let (_, origin) = self.client_window();
+        // Step 1
+        self.MoveTo(x + origin.x, y + origin.y)
+    }
+
+    // https://drafts.csswg.org/cssom-view/#dom-window-screenx
+    fn ScreenX(self) -> i32 {
+        let (_, origin) = self.client_window();
+        origin.x
+    }
+
+    // https://drafts.csswg.org/cssom-view/#dom-window-screeny
+    fn ScreenY(self) -> i32 {
+        let (_, origin) = self.client_window();
+        origin.y
+    }
+
+    // https://drafts.csswg.org/cssom-view/#dom-window-outerheight
+    fn OuterHeight(self) -> i32 {
+        let (size, _) = self.client_window();
+        size.height.to_i32().unwrap_or(1)
+    }
+
+    // https://drafts.csswg.org/cssom-view/#dom-window-outerwidth
+    fn OuterWidth(self) -> i32 {
+        let (size, _) = self.client_window();
+        size.width.to_i32().unwrap_or(1)
+    }
+
+    // https://drafts.csswg.org/cssom-view/#dom-window-devicepixelratio
+    fn DevicePixelRatio(self) -> Finite<f64> {
+        let dpr = self.window_size.get()
```

Comment:
This will return the wrong DPR if we are running with an option specifiying DPR. (See device_pixels_per_screen_px in compositor.rs: http://mxr.mozilla.org/servo/source/components/compositing/compositor.rs#1139)
Not sure how much we care though.